PostgreSQL: Enhance connection string handling

zoltanbedi opened the pull request:
**What is this feature?**

- Updated documentation to include various formats for specifying the Host URL, including examples for IPv4, IPv6, full PostgreSQL URLs, and Unix socket paths.
- Added a note regarding the use of credentials from the URL when respective fields are empty.
- Improved connection string generation in the PostgreSQL datasource to handle full URLs, including parsing for user, password, and database name.
- Added test cases for various connection string scenarios, including full URLs and error handling for invalid formats.

**Why do we need this feature?**

Many times people try to use full url of their connection strings in the host field.

**Who is this feature for?**

Grafana admins who set up postgresql plugins.

**Which issue(s) does this PR fix?**:

Fixes grafana/grafana-postgresql-datasource#143

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

zoltanbedi committed:
Enhance PostgreSQL datasource documentation and connection string handling

- Updated documentation to include various formats for specifying the Host URL, including examples for IPv4, IPv6, full PostgreSQL URLs, and Unix socket paths.
- Added a note regarding the use of credentials from the URL when respective fields are empty.
- Improved connection string generation in the PostgreSQL datasource to handle full URLs, including parsing for user, password, and database name.
- Added test cases for various connection string scenarios, including full URLs and error handling for invalid formats.

ivanahuckova commented on docs/sources/datasources/postgres/configure/_index.md:
```diff
@@ -89,8 +89,17 @@ Following is a list of PostgreSQL configuration options:
+  - Host with port: `localhost:5432`
+  - IPv4 address: `192.168.1.1:5432`
+  - IPv6 address: `[2001:db8::1]:5432`
+  - Full PostgreSQL URL: `postgresql://user:password@host:5432/dbname`
```
Do we want to add this as a last option and mention that this is not recommended as password is in this cased not saved as secure field?

ivanahuckova reviewed (commented):
Nice! LGTM, but could we mention in the docs that **Full PostgreSQL URL** is not recommended as in that case password is not stored securely.

zoltanbedi committed:
docs(postgres): add warning about including credentials in connection URL

- Added a note in the PostgreSQL configuration documentation highlighting that including the username and password in the connection URL is not recommended due to security concerns.

zoltanbedi committed:
Merge remote-tracking branch 'origin/main' into postgres-host-handling

ivanahuckova reviewed (approved):
LGTM

zoltanbedi committed:
Merge remote-tracking branch 'origin/main' into postgres-host-handling

yesoreyeram reviewed (approved):
LGTM. Once the `Unix socket path` documented properly I think it is good to go.

zoltanbedi commented:
I'm going to close this in favor of https://github.com/grafana/grafana/pull/107713 which adds support for copy pasting a connection string.
